### DISCUSSION TITLE
Form in request not displaying example in documentation

### DISCUSSION THREAD

**Discussion by yudataguy** [fastapi/fastapi#8429] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.post(
    "/face-quality",
    status_code=status.HTTP_200_OK,
    response_model=FaceQuality,
)
async def v1_face_quality_post(
    image: Optional[UploadFile] = File(None, description="Image file"),
    url: Optional[str] = Form(None, description="Image url", example="http://example.com/sample.jpg"),
    label: Optional[str] = Query(None, description="label", example="tom")
):
```

### Description

As you can see from the code, I use `Form` or `Query` in request, but for `Query`, the `example` display correctly in documentation, but not in `Form`, `description` both display fine. I have looked into the source code, where `example` field exists for `Form` class. Couldn't find much detail on the `example` field. Tried different format, but none works. 

Any idea what is the correct format for `example` field in `Form` class? 

### Operating System

Linux

### FastAPI Version

0.75.2

### Python Version

3.8.10

**Comment by seongwonhan88** [comment, marked as answer]
Hi @yudataguy, 
I think I see your issue and it's because it's not declared under pydantic schema.

If you read the doc [here](https://fastapi.tiangolo.com/tutorial/schema-extra-example/#technical-details), and it says 
>  But when you use example or examples with any of the other utilities (Query(), Body(), etc.) those examples are not added to the JSON Schema that describes that data (not even to OpenAPI's own version of JSON Schema), they are added directly to the path operation declaration in OpenAPI (outside the parts of OpenAPI that use JSON Schema).

Instead of tying on the function argument, I would change this to something like

```diff 
- async def v1_face_quality_post(
-    image: Optional[UploadFile] = File(None, description="Image file"),
-   url: Optional[str] = Form(None, description="Image url", example="http://example.com/sample.jpg"),
-   label: Optional[str] = Query(None, description="label", example="tom")
- ):

+ class PostSchema(BaseModel):
+     image: Optional[UploadFile] = Field(description="Image file")
+     url: Optional[str] = Field(description="image url", example="http://example.com/sample.jpg")
+     label: Optional[str] = Field(description="label", example="tom")

+ async def v1_face_quality_post(data: PostSchema): 
+      ...

```

You can also provide more detailed example like [here](https://fastapi.tiangolo.com/tutorial/schema-extra-example/#pydantic-schema_extra)

```py
class Item(BaseModel):
    name: str
    description: Union[str, None] = None
    price: float
    tax: Union[float, None] = None

    class Config:
        schema_extra = {
            "example": {
                "name": "Foo",
                "description": "A very nice Item",
                "price": 35.4,
                "tax": 3.2,
            }
        }
```
I hope you find it helpful.
